sdk/eventhub/Azure.Messaging.EventHubs/stress/src/Infrastructure/BufferedPublisher.cs

```diff
@@ -0,0 +1,155 @@
+
+                        if (testConfiguration.ConcurrentSends > 1)
+                        {
+                            for (var index = 0; index < testConfiguration.ConcurrentSends - 1; ++index)
+                            {
+                                enqueueTasks.Add(Task.Run(async () =>
+                                {
+                                    while (!cancellationToken.IsCancellationRequested)
+                                    {
+                                        await PerformSend(producer, cancellationToken).ConfigureAwait(false);
+
+                                        if ((testConfiguration.ProducerPublishingDelay.HasValue) && (testConfiguration.ProducerPublishingDelay.Value > TimeSpan.Zero))
+                                        {
+                                            await Task.Delay(testConfiguration.ProducerPublishingDelay.Value, backgroundCancellationSource.Token).ConfigureAwait(false);
+                                        }
+                                    }
+                                }));
+                            }
+                        }
+
+                        // Perform one of the sends in the foreground, which will allow easier detection of a
+                        // processor-level issue.
+
+                        while (!cancellationToken.IsCancellationRequested)
+                        {
+                            try
+                            {
+                                await PerformSend(producer, cancellationToken).ConfigureAwait(false);
+
+                                if ((testConfiguration.ProducerPublishingDelay.HasValue) && (testConfiguration.ProducerPublishingDelay.Value > TimeSpan.Zero))
+                                {
+                                    await Task.Delay(testConfiguration.ProducerPublishingDelay.Value, cancellationToken).ConfigureAwait(false);
+                                }
+                            }
+                            catch (TaskCanceledException)
+                            {
+                                backgroundCancellationSource.Cancel();
+                                await Task.WhenAll(enqueueTasks).ConfigureAwait(false);
+                            }
+                        }
+                    }
+                }
+                catch (TaskCanceledException)
+                {
+                    // No action needed.
+                }
+                catch (Exception ex) when
+                    (ex is OutOfMemoryException
+                    || ex is StackOverflowException
+                    || ex is ThreadAbortException)
+                {
+                    throw;
+                }
+                catch (Exception ex)
+                {
+                    metrics.Client.TrackException(ex);
+                }
+            }
+        }
+
+        private async Task PerformSend(EventHubBufferedProducerClient producer,
```

Comment:
nit:  We probably should rename this to denote that it's enqueuing rather than sending.